src/hera/cron_workflow.py

```diff
@@ -55,11 +59,14 @@ def __init__(
         service_account_name: Optional[str] = None,
         labels: Optional[Dict[str, str]] = None,
         namespace: Optional[str] = 'default',
+        use_unique_name: Optional[bool] = True,
     ):
         if timezone and timezone not in pytz.all_timezones:
             raise ValueError(f'{timezone} is not a valid timezone')
-
-        self.name = f'{name.replace("_", "-")}-{str(uuid4()).split("-")[0]}'
+        self.name = name.replace("_", "-")
+        self.unique_name = f'{self.name}-{str(uuid4()).split("-")[0]}'
+        if use_unique_name:
+            self.name = self.unique_name
```

Comment:
It strikes me that 
```python
self.name = name.replace("_",  "-")    # RFC1123
if use_unique_name:
    self.name = "-".join([self.name, str(uuid4()).split("-")[0]])
```
is a _bit_ clearer. Thoughts? 🙂

Reply:
Hi @flaviuvadan  - It's clearer but it doesn't preserve the unique_name for use in the template.  I found that when I overrode the name, metadata.name and template.name I got an error when trying to create the workflow.  I had to leave the template.name with the generated unique name to avoid the error.  I can look into that again and see what the specific error was.